app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -1088,6 +1092,17 @@ class WebViewActivity : BaseActivity(), io.homeassistant.companion.android.webvi
             return
         isShowingError = true
 
+        // If the endpoint requires a TLS client certificate and we got a timeout, ignore the error
+        // It is likely due to the user not choosing a key yet
+        val tlsWebViewclient = webView.webViewClient as TLSWebViewClient
```

Comment:
What would be the best way to handle this? Check for the API version and if below 26 fallback on a standard `WebViewClient`?

Reply:
As far as I understand these API-level things, what matters is to never allow the execution to reach a point where a class/method is used that is not present on the device's API level.
I think `TLSWebViewClient::onReceivedClientCertRequest` is OK, and doesn't need modifications (but pls confirm @JBassett).

For `WebViewActivity`, I think all we need is an if around the `TLSWebViewClient`-related part:
```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
    // If the endpoint requires a TLS client certificate and we got a timeout, ignore the error
    // It is likely due to the user not choosing a key yet
    val tlsWebViewclient = webView.webViewClient as TLSWebViewClient
    if (tlsWebViewclient != null &&
        tlsWebViewclient.isTLSClientAuthNeeded &&
        !tlsWebViewclient.hasUserDeniedAccess &&
        errorType == ErrorType.TIMEOUT
    ) {
        return
    }
}
```

Reply:
update: I pasted the code part I wrote in my previous comment into Android Studio, and it showed me a warning telling me there's a compatibility layer util method for this: `WebViewCompat.getWebViewClient(webView)`

Here's how it could be used:
(disclaimer: I didn't try to run the code, I'm posting this blindly)

```
var tlsWebViewclient: TLSWebViewClient? = null
// If the endpoint requires a TLS client certificate and we got a timeout, ignore the error
// It is likely due to the user not choosing a key yet
if (WebViewFeature.isFeatureSupported(WebViewFeature.GET_WEB_VIEW_CLIENT)) {
    tlsWebViewclient = WebViewCompat.getWebViewClient(webView) as TLSWebViewClient
    if (tlsWebViewclient.isTLSClientAuthNeeded &&
        !tlsWebViewclient.hasUserDeniedAccess &&
        errorType == ErrorType.TIMEOUT
    ) {
        return
    }
}
```

And this way there's no need for checking the API level.